plugins/inputs/pubsub/README.md

```diff
@@ -0,0 +1,86 @@
+# Google Cloud PubSub Input Plugin
+
+The GCP PubSub plugin ingests metrics from [Google Cloud PubSub][pubsub]
+and creates metrics using one of the supported [input data formats][].
+
+
+### Configuration
+
+This section contains the default TOML to configure the plugin.  You can
+generate it using `telegraf --usage pubsub`.
+
+```toml
+[[inputs.pubsub]]
+  ## Name of Google Cloud Platform (GCP) Project owning PubSub subscriptions
+  google_project = "my-project"
+
+  ## Name of PubSub subscription
+  subscription = "my-subscription"
+
+  ## Optional filepath for GCP credentials JSON file to authorize calls to 
+  ## PubSub APIs. If not set explicitly, Telegraf will attempt to use 
+  ## Application Default Credentials, which is preferred. 
+  # credentials_file = "path/to/my/creds.json"
+
+  ## Maximum byte length of a message to consume. Larger messages are dropped.
+  ## If less than 0 or unspecified, defaults to unlimited length.
+  max_message_len = 1000000
+
+  ## Maximum messages to read from PubSub that have not been written to an
+  ## output.  For best throughput set based on the number of metrics within
+  ## each message and the size of the output's metric_batch_size.
+  ##
+  ## For example, if each message contains 10 metrics and the output 
+  ## metric_batch_size is 1000, setting this to 100 will ensure that a
+  ## full batch is collected and the write is triggered immediately without
+  ## waiting until the next flush_interval.
+  # max_undelivered_messages = 1000
+
+  ## The following are optional Subscription ReceiveSettings in PubSub.
+  ## Read more about these values:
+  ## https://godoc.org/cloud.google.com/go/pubsub#ReceiveSettings
+  
+  ## Maximum number of seconds for which a PubSub subscription
+  ## should auto-extend the PubSub ACK deadline for each message. If less than
+  ## 0, auto-extension is disabled.
+  # max_extension = 0
+
+  ## Maximum number of unprocessed messages in PubSub (unacknowledged but not 
+  ## yet expired in PubSub). A value of 0 is treated as the default 
+  ## PubSub value. Negative values will be treated as unlimited.
+  # max_outstanding_messages = 0
```

Comment:
yeah, i think it might make sense if you want to pipe more through telegraf but are limited by the default amount in PubSub, so you just set this value to very large and then manage it in Telegraf instead? This also might be a "it's there so might as well" situation